### PR TITLE
TST #24444 added tests

### LINKED ISSUE #24444: Imprecision when converting Period to timestamp
#### Code Sample, a copy-pastable example if possible

```python
pd.Period("2000-12-31 23:59:59.999").to_timestamp()
```
returns: 2000-12-31 23:59:59.998999999
instead of 2000-12-31 23:59:59.999
It works correctly for other dates, e.g.
```python
pd.Period("2000-12-31 23:59:59.99").to_timestamp()
```
returns correctly "2000-12-31 23:59:59.990000"

#### Problem description
incorrect conversion of Period to Timestamp, see above.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None

pandas: 0.23.4
pytest: None
pip: 18.1
setuptools: 39.1.0
Cython: None
numpy: 1.15.4
scipy: 1.2.0
pyarrow: None
xarray: None
IPython: 7.2.0
sphinx: None
patsy: 0.5.1
dateutil: 2.7.5
pytz: 2018.7
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 3.0.2
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by OlivierLuG**
At last, there are no issues at all. I've made a new PR.

- [x] closes #24444
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
Thanks @OlivierLuG 

> I've made a new PR to update master. Sorry, there iare conflict. I don't know how to handle that.

There's instructions [here](https://pandas.pydata.org/pandas-docs/stable/development/contributing.html#updating-your-pull-request) in the contributing guide

File: pandas/tests/scalar/period/test_period.py [hunk 1]
```diff
@@ -649,6 +649,26 @@ def test_to_timestamp_business_end(self):
...
+            ("1999-12-31 23:59:59.999999", 999999),
+            ("2050-12-31 23:59:59.5", 500000),
+            ("2050-12-31 23:59:59.500001", 500001),
+            ("2050-12-31 23:59:59.123456", 123456),
+        ],
+    )
+    @pytest.mark.parametrize("freq", [None, "us", "ns"])
+    def test_to_timestamp_microsecond(self, ts, expected, freq):
+        # GH 24444
+        result = Period(f"{ts}").to_timestamp(freq=freq).microsecond
```
Comment (MarcoGorelli) [inline pandas/tests/scalar/period/test_period.py:669]: Is the f-string necessary here?
```suggestion
        result = Period(ts).to_timestamp(freq=freq).microsecond
```

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

File: pandas/tests/scalar/period/test_period.py (hunk 1 above)
Comment (OlivierLuG) [reply pandas/tests/scalar/period/test_period.py]: It is not !, a new PR was sent.

**Review by WillAyd — APPROVED** [review]
lgtm

**Comment by mroeschke** [comment]
Thanks @OlivierLuG

**Pull request merged**